src/utils/telemetry.ts

```diff
@@ -19,6 +24,35 @@ export const trackError = (error: any) => {
   }
 };
 
+export type Event = keyof EventMap;
+export type EventMap = {
+  CREATE_EMPTY_PROJECT: undefined;
+  SELECT_LEFT_TOOL_PANEL: {panelID: string};
+  START_FROM_A_TEMPLATE: {templateID: string};
+  WINDOW_HELP_LINK: {linkID: string};
+  USE_TEMPLATE: {templateID: string};
+  DO_HELM_PREVIEW: undefined;
+  CLUSTER_VIEW: {numberOfResourcesInCluster: number};
+  DO_KUSTOMIZE_PREVIEW: undefined;
+  OPEN_EXISTING_PROJECT: {numberOfFiles: number; numberOfResources: number};
+  ADD_NEW_RESOURCE: {resourceKind: string};
+  APP_INSTALLED: {appVersion: string};
+  CHANGES_BY_FORM_EDITOR: {resourceKind?: string};
+  CHANGES_BY_SETTINGS_PANEL: {type: string; settingKey: string};
+  APPLY: {kind: string} & Record<string, any>;
+  DIFF: undefined;
+  CLUSTER_COMPARE: {numberOfResourcesBeingCompared: number} | {fail: string};
+  FOLLOW_LINK: {type: string};
+  QUICK_SEARCH: undefined;
+  UPDATE_APPLICATION: undefined;
+  APPLY_FILE: undefined;
+  APPLY_HELM_CHART: undefined;
+  RUN_PREVIEW_CONFIGURATION: undefined;
+  VALIDATION_PANE_OPENED: {id: ValidationIntegrationId};
+  OPA_ENABLED: {all: boolean};
+  OPA_DISABLED: {all: boolean};
```

Comment:
I wonder if our analytics platform works well enough to work with these kind of parametrised events or whether we need to split it up in OPA_ENABLED and OPA_ALL_ENABLED.

Reply:
it works well, we can send any object as a payload as long as it doesn't contain functions